cpp/src/io/orc/timezone.cpp

```diff
@@ -230,17 +230,18 @@ void posix_parser<Container>::skip_name()
 {
   CUDF_EXPECTS(cur < end, "Unexpected end of input stream");
 
-  auto c = *cur;
-  if (c == '<') {
-    ++cur;
-    while (cur < end) {
-      if (*cur++ == '>') { break; }
-    }
-  } else {
-    while ((c < '0' || c > '9') && (c != '-') && (c != '+') && (c != ',')) {
-      if (++cur >= end) { break; }
-      c = *cur;
+  while (cur < end && (*cur < '0' || *cur > '9') && (*cur != '-') && (*cur != '+') &&
```

Comment:
This looks like a place where `std::regex` would provide a solution that is a lot cleaner and easier to understand.

Reply:
Would regex be slower? Looks like we're executing this for each character.

Reply:
I have to admit I have no idea how to implement this using regex. Any pointers would be appreciated.

Reply:
So it looks like we're trying to advance the internal `cur` iterator past a particular "name" pattern. What does the name look like?

Reply:
The pattern can contain non-digit characters, and/or any characters within '<>'. As an example, the input I found to be broken before this PR is "\n<3>". The other type of names I've seen is something like "PST".

Reply:
I don't think my regex-fu is going to be strong enough to replace what is here. My main point is that the existing logic is pretty  unwieldy and difficult to parse, so it's not surprising that there was a bug. I was hoping to find a way that better expresses the intent of what's going on, but don't block the PR because of this.

Reply:
This is what I have:
```cpp
  cur = std::find_if(cur, end, [](auto c) { 
    return std::isdigit(c) || c == '-' || c == ',' || c == '+' || c == '<'; 
  });

  if (*cur == '<') cur = std::next(std::find(cur, end, '>'));
```

Reply:
This was the [godbolt](https://www.godbolt.org/z/cbz8nW).

Reply:
Nice! But, these are not equivalent as the original code also skips past any non-digit characters after '>'. I'll try to find the exact specs for this field.

Reply:
Finally found it spelled out, outside of the official specs: https://www.postgresql.org/docs/13/datetime-posix-timezone-specs.html
> a zone abbreviation can be a string of letters, such as EST, or an arbitrary string surrounded by angle brackets, such as `<UTC-05>`

So @codereport 's implementation covers it. Thanks!